stress/cpuset_manager.go

```diff
@@ -78,7 +81,28 @@ func (manager *cpuStressserManager) TrackInjectorCores(cgroup cgroup.Manager) (c
 		return cpuset.NewCPUSet(), fmt.Errorf("error parsing cpuset allocated cores: %w", err)
 	}
 
-	manager.setCoresToBeStressed(manager.coresToBeStressed.Union(cores))
+	coresToBeStressedCount, _ := manager.userRequestedCores(cores.Size(), userRequestCount)
+	coresToBeStressed := cpuset.NewCPUSet(cores.ToSlice()[:coresToBeStressedCount]...)
+
+	manager.log.Infof("retrieved coresToBeStressed after applying userRequestCount %s: %s", userRequestCount.StrVal, coresToBeStressed)
+	manager.setCoresToBeStressed(manager.coresToBeStressed.Union(coresToBeStressed))
 
 	return cores, nil
 }
+
+func (manager *cpuStressserManager) userRequestedCores(cores int, userRequestedCores *intstr.IntOrString) (int, error) {
+	targetCores := 0
+	userRequestedCount, isPercent, err := v1beta1.GetIntOrPercentValueSafely(userRequestedCores)
```

Comment:
Are we able to use the method from `apimachinery` somehow:
* https://github.com/kubernetes/apimachinery/blob/master/pkg/util/intstr/intstr.go#L150

Seems they are doing what you are?

Should we also get rid of our own method if possible?

Also the method I suggested seems to already exists in our helpers:
* https://github.com/DataDog/chaos-controller/blob/main/controllers/helpers.go#L43

Maybe using this directly would be good enough if we do not intend to use `apimachinery` or can't

Reply:
nice, I didn't discover `helpers.go`, I will see to use this now. Thanks!